api/lorawan.proto

```diff
@@ -254,24 +254,26 @@ message TxSettings {
   string coding_rate = 3;
   // Frequency (Hz).
   uint64 frequency = 4;
-  // Transmission power (dBm). Only on downlink.
-  float tx_power = 5;
-  // Invert LoRa polarization; false for LoRaWAN uplink, true for downlink.
-  bool invert_polarization = 6;
-  // Index of the gateway channel that received the message.
-  // Set by Gateway Server.
-  uint32 gateway_channel_index = 7 [(validate.rules).uint32 = {lte: 255}];
-  // Index of the device channel that received the message.
-  // Set by Network Server.
-  uint32 device_channel_index = 8 [(validate.rules).uint32 = {lte: 255}];
   // Send a CRC in the packet; only on uplink; on downlink, CRC should not be enabled.
-  bool enable_crc = 9 [(gogoproto.customname) = "EnableCRC"];
+  bool enable_crc = 5 [(gogoproto.customname) = "EnableCRC"];
   // Timestamp of the gateway concentrator when the uplink message was received, or when the downlink message should be transmitted (microseconds).
   // On downlink, set timestamp to 0 and time to null to use immediate scheduling.
-  uint32 timestamp = 10;
+  uint32 timestamp = 6;
   // Time of the gateway when the uplink message was received, or when the downlink message should be transmitted.
   // For downlink, this requires the gateway to have GPS time synchronization.
-  google.protobuf.Timestamp time = 11 [(gogoproto.stdtime) = true];
+  google.protobuf.Timestamp time = 7 [(gogoproto.stdtime) = true];
+  // Fields that are specific for transmission on a gateway 
+  Downlink downlink = 8;
+
+  // Fields that are specific for transmission on a gateway
```

Comment:
- `Transmission settings for downlink.`
- Declare before use of field, like other embedded messages

Reply:
Actually I used the `message MACCommand` as reference on `L366`